pkg/compose/compose_test: TestComposeCompare failed

pkg/compose/compose_test.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/20589737?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/20589737?buildTab=artifacts#/) on master @ [44eef4bdf994db9b8fcc5149c0e534f1f1c6fd7a](https://github.com/cockroachdb/cockroach/commits/44eef4bdf994db9b8fcc5149c0e534f1f1c6fd7a):


```
        cockroach1-1  | sql (JDBC):          jdbc:postgresql://cockroach1:26257/defaultdb?sslmode=disable&user=root
        cockroach1-1  | RPC client flags:    /cockroach/cockroach <client cmd> --host=cockroach1:26257 --insecure
        cockroach1-1  | logs:                /cockroach/store/logs
        cockroach1-1  | temp dir:            /cockroach/store/cockroach-temp3378937119
        cockroach1-1  | external I/O path:   /cockroach/store/extern
        cockroach1-1  | store[0]:            path=/cockroach/store
        cockroach1-1  | clusterID:           ca3764f7-8c44-4ff5-997d-9e68aba44928
        cockroach1-1  | status:              initialized new cluster
        cockroach1-1  | nodeID:              1
        test-1        |     compare_test.go:109: Checking connection to: cockroach2
        test-1        | === RUN   TestCompare/mutators
        test-1        |     compare_test.go:118: starting test: mutators
        test-1        |     compare_test.go:127: initializing connection: cockroach1
        test-1        |     compare_test.go:127: initializing connection: cockroach2
        test-1        |     compare_test.go:189: DO $$
        test-1        |         DECLARE
        test-1        |         decl_67231 INTERVAL := '-57 years -5 mons -315 days -16:03:54.105476':::INTERVAL;
        test-1        |         decl_67232 TSQUERY := '''uxyvU'' <-> !''ado'' | ''kXgNF''':::TSQUERY;
        test-1        |         decl_67233 rand_typ_0;
        test-1        |         decl_67234 STRING;
        test-1        |         decl_67235 REGROLE;
        test-1        |         BEGIN
        test-1        |         decl_67235 := decl_67235;
        test-1        |         NULL;
        test-1        |         NULL;
        test-1        |         WHILE true LOOP
        test-1        |         decl_67234 := NULL;
        test-1        |         decl_67235 := decl_67235;
        test-1        |         SELECT '[-0.04532914,2.015718,0.36568788,0.32227784,-1.4928008,-0.83328277,-0.15317434,1.0307009,-0.6408299,0.8670871,1.5956627,0.62583476,2.0204067,0.20311098,0.90798664,-0.46172184,0.18768688,0.44258988,-0.06572735,-1.8810806,-0.3812781,0.4360722,-0.9637413,-0.42673305,0.11297186,2.0730221,-0.62356925,-1.2264546,-0.5886528,-0.6831411,-0.1379659,0.8502259,1.4854848,-0.44312596,-0.4310865,0.43922877,-0.3085667,0.8631196,0.0871788,0.6771113,1.3131676,-1.7922107,1.9299598,-1.7752051,1.680124,1.8024857,0.05230585,-0.71980685]':::VECTOR AS col_439154, decl_67232 AS col_439155, (SELECT (-3.830658752581626482E+33):::DECIMAL AS "c'ol_439156" ORDER BY "c'ol_439156" DESC LIMIT 1:::INT8) AS "%pco\\U000B94E5l_439157", '2(^-G':::STRING AS "COl̦_439158", "ta b_218144".col3_3 AS col_439159, "ta b_218144".col3_6 AS col_439160, 0:::OID AS "
col_439161", B'1011111010110001000111' AS col_439162, "ta b_218144"."
        test-1        |         col' 3_0" AS col_439163, 1564882960:::REGROLE AS "'%q�col_439164", (-2590412.036053356706):::DECIMAL AS col_439165, 3368896727:::REGROLE AS "co-l_439166", '':::STRING AS col_439167, decl_67233 AS co😋l_439168, "ta b_218144"."
        test-1        |         col' 3_0" AS """col_439169", NULL AS col_439170 FROM postgres.public.table_3 AS "ta b_218144" ORDER BY "ta b_218144".col3_5 ASC;
        test-1        |         END LOOP;
        test-1        |         decl_67234 := decl_67234;
        test-1        |         END;
        test-1        |         $$;
        test-1        |     compare_test.go:190: cockroach1: ping: conn closed
        test-1        | --- FAIL: TestCompare (267.97s)
        test-1        |     --- FAIL: TestCompare/mutators (266.79s)
        test-1        | FAIL
        
�[Ktest-1 exited with code 1
        Aborting on container exit...
         Container compare-test-1  Stopping
         Container compare-test-1  Stopped
         Container compare-cockroach2-1  Stopping
         Container compare-cockroach1-1  Stopping
         Container compare-cockroach2-1  Stopped
         Container compare-cockroach1-1  Stopped
        
    compose_test.go:160: exit status 1
--- FAIL: TestComposeCompare (285.39s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #153083 pkg/compose/compose_test: TestComposeCompare failed [connection closed] [C-test-failure O-robot P-2 T-sql-foundations branch-release-24.1.24-rc]
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-55318

--- a/pkg/compose/compare/compare/compare_test.go
+++ b/pkg/compose/compare/compare/compare_test.go
@@ -79,8 +79,13 @@ func TestCompare(t *testing.T) {
 	}
 	configs := map[string]testConfig{
 		"mutators": {
-			setup:           sqlsmith.Setups[sqlsmith.RandTableSetupName],
-			opts:            []sqlsmith.SmitherOption{sqlsmith.CompareMode()},
+			setup: sqlsmith.Setups[sqlsmith.RandTableSetupName],
+			opts: []sqlsmith.SmitherOption{
+				sqlsmith.CompareMode(),
+				// TODO(yuzefovich): perhaps allow DO blocks again after they
+				// have been stabilized in other tests.
+				sqlsmith.DisableDoBlocks(),
+			},
 			ignoreSQLErrors: true,
 			conns: []testConn{
 				{

--- a/pkg/sql/routine.go
+++ b/pkg/sql/routine.go
@@ -163,7 +163,7 @@ func (p *planner) EvalRoutineExpr(
 		if routineDepthValue := ctx.Value(routineDepthKey{}); routineDepthValue != nil {
 			routineDepth = routineDepthValue.(int)
 		}
-		const maxDepth = 10000
+		const maxDepth = 100
 		if routineDepth > maxDepth {
 			return nil, pgerror.Newf(pgcode.ProgramLimitExceeded,
 				"routine reached recursion depth limit: %d (probably infinite loop)", maxDepth)

